optimizely/client/client.go

```diff
@@ -42,6 +42,41 @@ type OptimizelyClient struct {
 	executionCtx utils.ExecutionCtx
 }
 
+// Activate returns the key of the variation the user is bucketed into and sends an impression event to the Optimizely log endpoint
+func (o *OptimizelyClient) Activate(experimentKey string, userContext entities.UserContext) (result string, err error) {
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch t := r.(type) {
+			case error:
+				err = t
+			case string:
+				err = errors.New(t)
+			default:
+				err = errors.New("unexpected error")
+			}
+			errorMessage := fmt.Sprintf("optimizely SDK is panicking with the error:")
+			logger.Error(errorMessage, err)
+			logger.Debug(string(debug.Stack()))
+		}
+	}()
+
+	decisionContext, experimentDecision, err := o.getExperimentDecision(experimentKey, userContext)
+	if err != nil {
+		logger.Error("received an error while computing experiment decision", err)
+		return result, err
+	}
+
+	if experimentDecision.Variation != nil {
```

Comment:
can you have nil error and variation still nil ?

Reply:
Yep, what @mjc1283 said